blocks/example-file-block/index.tsx

```diff
@@ -74,10 +74,15 @@ type Props = {
   metadata: any;
 } & (FileData | FolderData);
 
-const root = ReactDOM.createRoot(document.getElementById("root")!);
-
 function setProps(props: Props, blocksAPI: BlocksAPI) {
-  root.render(<BlockComponent {...props} {...blocksAPI} />);
+  let root: ReactDOM.Root | undefined;
+  let component = <BlockComponent {...props} {...blocksAPI} />;
+  if (root) {
+    root.render(component);
+  } else {
```

Comment:
I think this will never run, since `let root` is scoped to the function and will be recreated every run. You could do it with a `let` outside the function (but I'm not sure what happens to these if the code is reloaded).

Reply:
Curious, that's not what I was observing on my end.

Closing out this patch in any case!

Reply:
ok I did not actually test it, you may be right! (@Wattenberger and I ended up stripping out the React code here anyway, not sure if she pushed the work we did Friday)